packages/block/src/block.ts

```diff
@@ -82,455 +56,6 @@ export class Block {
-      : undefined
-
-    const hasDepositRequests = depositRequests !== undefined && depositRequests !== null
-    const hasWithdrawalRequests = withdrawalRequests !== undefined && withdrawalRequests !== null
-    const hasConsolidationRequests =
-      consolidationRequests !== undefined && consolidationRequests !== null
-
-    const requests =
-      hasDepositRequests || hasWithdrawalRequests || hasConsolidationRequests
-        ? ([] as CLRequest<CLRequestType>[])
-        : undefined
-
-    if (depositRequests !== undefined && depositRequests !== null) {
-      for (const dJson of depositRequests) {
-        requests!.push(DepositRequest.fromJSON(dJson))
-      }
-    }
-    if (withdrawalRequests !== undefined && withdrawalRequests !== null) {
-      for (const wJson of withdrawalRequests) {
-        requests!.push(WithdrawalRequest.fromJSON(wJson))
-      }
-    }
-    if (consolidationRequests !== undefined && consolidationRequests !== null) {
-      for (const cJson of consolidationRequests) {
-        requests!.push(ConsolidationRequest.fromJSON(cJson))
-      }
-    }
-
-    const requestsRoot = requests
-      ? await Block.genRequestsTrieRoot(requests, new Trie({ common: opts?.common }))
-      : undefined
-
-    const header: HeaderData = {
-      ...payload,
-      number,
-      receiptTrie,
-      transactionsTrie,
-      withdrawalsRoot,
-      mixHash,
-      coinbase,
-      requestsRoot,
-    }
-
-    // we are not setting setHardfork as common is already set to the correct hf
-    const block = Block.fromBlockData(
-      { header, transactions: txs, withdrawals, executionWitness, requests },
-      opts
-    )
-    if (
-      block.common.isActivatedEIP(6800) &&
-      (executionWitness === undefined || executionWitness === null)
-    ) {
-      throw Error('Missing executionWitness for EIP-6800 activated executionPayload')
-    }
-    // Verify blockHash matches payload
-    if (!equalsBytes(block.hash(), hexToBytes(payload.blockHash as PrefixedHexString))) {
-      const validationError = `Invalid blockHash, expected: ${
-        payload.blockHash
-      }, received: ${bytesToHex(block.hash())}`
-      throw Error(validationError)
-    }
-
-    return block
-  }
-
-  /**
-   *  Method to retrieve a block from a beacon payload json
-   * @param payload json of a beacon beacon fetched from beacon apis
-   * @param opts {@link BlockOptions}
-   * @returns the block constructed block
-   */
-  public static async fromBeaconPayloadJson(
-    payload: BeaconPayloadJson,
-    opts?: BlockOptions
-  ): Promise<Block> {
-    const executionPayload = executionPayloadFromBeaconPayload(payload)
-    return Block.fromExecutionPayload(executionPayload, opts)
-  }
-
   /**
    * This constructor takes the values, validates them, assigns them and freezes the object.
    * Use the static factory methods to assist in creating a Block object from varying data types and options.
```

Comment:
One thing which has gotten a bit worse with this extraction is discoverability, so it has gotten harder to find/auto-complete what constructors are available.

Can we make it a habit to therefore to always provide a list with all the (linked, so clickable) static constructor methods, either here or along the class (`Block`) code docs (can't fully decide, where does one stumble more upon this? If one imports a `Block` and then hovers over the class import or if one types in `new Block()` and then sees the code docs for the constructor? Guess maybe the class docs?).

So that would be great and pretty helpful!

Also: can you add a `@deprecated` note here for this constructor as well, like we have for other constructors (tx? evm?) already?

Reply:
I agree.  It is a nicer experience to type in `Block.` and have the constructors pop up.